### PR TITLE
Purge 'side' parameter

### PR DESCRIPTION
**Pull request by PetervDooren**
We dont want to refer to left or right arms because not every robot has two arms which are symetrical. For that reason we want to remove the parameter 'side' and replace it with something more general

Must be merged together with: https://github.com/tue-robotics/hero_description/pull/11 and https://github.com/tue-robotics/hero_bridge/pull/23

I think this should be tested on the real robot, just to be safe...

### CODE REVIEW

File: robot_skills/src/robot_skills/arm/arms.py
Comment (PetervDooren) [inline robot_skills/src/robot_skills/arm/arms.py:279]: To make this interface truly generic we should make it so that not all parameters, actions, tf_frames are linked via the arm name. Then we should allow these things as parameters to the interface. But that might be something for another day...

**Review by MatthijsBurgh — DISMISSED** [review]

File: robot_skills/package.xml
Comment (MatthijsBurgh) [inline robot_skills/package.xml:44]: I think this is a very dirty dependency. But this might be fixed by @jlunenburg.

File: robot_skills/src/robot_skills/arm/arms.py
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arm/arms.py:289]: Why have `name` as argument, but store it as `arm_name`. I think it is better to use `name` inside the class.

File: robot_skills/src/robot_skills/arm/gripper.py
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arm/gripper.py:71]: I would say a gripper is part of an arm, so the parameter is also part in the parameter tree of the arm.

File: robot_skills/test/rostest/test_hero.py
Comment (MatthijsBurgh) [inline robot_skills/test/rostest/test_hero.py:14]: I think it is common to do this in the `__main__`

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]

File: robot_skills/package.xml
Comment (PetervDooren) [reply robot_skills/package.xml:44]: It is dirty. But it is a dependency that we have always had implicitly as we cant construct a robot object without its description.

File: robot_skills/src/robot_skills/arm/arms.py
Comment (PetervDooren) [reply robot_skills/src/robot_skills/arm/arms.py:289]: Agreed

File: robot_skills/src/robot_skills/arm/gripper.py
Comment (PetervDooren) [reply robot_skills/src/robot_skills/arm/gripper.py:71]: It is not necessarily part of an arm. That is why we created a more composable robot interface

File: robot_skills/src/robot_skills/arm/arms.py
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arm/arms.py:296]: Shouldn't this be gripper specific

File: robot_skills/src/robot_skills/arm/grasp_sensor.py
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arm/grasp_sensor.py:105]: Now we force the robot name to be in the topic. Probably that is always the case, but I prefer we don't apply any unesesarry contraints on the topic. Let the user define the entire topic.

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]

File: robot_skills/src/robot_skills/arm/arms.py
Comment (PetervDooren) [reply robot_skills/src/robot_skills/arm/arms.py:296]: It should be, but it is outside the scope of this PR

File: robot_skills/src/robot_skills/arm/grasp_sensor.py
Comment (PetervDooren) [reply robot_skills/src/robot_skills/arm/grasp_sensor.py:105]: Almost all topics within the interface are prefixed with the robots name. I think it would cause more trouble than its worth at this stage. There are limits to how generic we want to make these interfaces. And being able to quickly change the prefix is an important part of making an interface for a new robot.

File: robot_skills/src/robot_skills/arm/arms.py
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/arm/arms.py:325]: This shouldn't be in the code.

File: robot_skills/src/robot_skills/arm/grasp_sensor.py
Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/arm/grasp_sensor.py:105]: But I am pretty sure that other robot parts don't do this. So please keep it consistent.

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]

File: robot_skills/src/robot_skills/arm/arms.py
Comment (PetervDooren) [reply robot_skills/src/robot_skills/arm/arms.py:325]: Yes will update the rviz config and remove this

Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/arm/arms.py:325]: I suggest to edit the config in an editor instead of rviz. This prevents any changes somewhere else in the config.

File: robot_skills/src/robot_skills/arm/grasp_sensor.py
Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/arm/grasp_sensor.py:105]: @PetervDooren I don't think we do this somewhere else. So therefore I still think we should change this.

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]
Only one remaining conversation

**Comment by PetervDooren** [comment]
I don't think the the fact that we dont do this like that anywhere else is reason to change this. 
I assume by 'this' you mean having a sensor topic as input?

**Comment by MatthijsBurgh** [comment]
I don't mind having the sensor topic as argument. I only mind that we force a prefix of the robot name on the topic.

**Comment by PetervDooren** [comment]
That is something we do with every topic/action/service. That is not something unusual in robot_skills

**Comment by MatthijsBurgh** [comment]
No, we don't do this in `robot_skills`. Not for the ones that have the topic as the argument. I.E. https://github.com/tue-robotics/tue_robocup/blob/9bece905c6b661a9c4f95d8d6dde2c6b5c203531/robot_skills/src/robot_skills/perception.py#L23-L31

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]
Please rebase before approval

File: robot_skills/package.xml
Comment (MatthijsBurgh) [inline robot_skills/package.xml:50]: I think this one can be removed now with the seperated skills

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]

File: robot_skills/package.xml
Comment (PetervDooren) [reply robot_skills/package.xml:50]: sharp, I missed that one

Comment (PetervDooren) [reply robot_skills/package.xml:50]:
```suggestion
```

**Review by MatthijsBurgh — APPROVED** [review]
Wait for CI before merging

File: robot_skills/src/robot_skills/torso.py
Comment (jlunenburg) [inline robot_skills/src/robot_skills/torso.py:32]: Why not load them outside and provide them to the torso via the constructor?

This solution isn't very nice because of two reasons:
* the try-except is error-prone. I'm not sure what'll happen if you move the torso
* it makes the torso depend on the fact that your arm is called 'arm_center', which is specific to Hero. So while making the Hero object more generic by removing 'left', you make this more specific by adding 'arm_center'

Comment (PetervDooren) [reply robot_skills/src/robot_skills/torso.py:32]: agreed, but this is a hotfix for hero anyway. The only change made here is that the dependency on some arm is made explicit.
A different PR should make this parameter no longer necessary in the first place. So that we can send only one joint goal while keeping all others the same.

Comment (jlunenburg) [reply robot_skills/src/robot_skills/torso.py:32]: Can I see that :blush: ? I like keeping the master branch as clean as possible and these hotfixes do not contribute to that.

Comment (PetervDooren) [reply robot_skills/src/robot_skills/torso.py:32]: You are right. I will make this cleaner. Probably by allowing arm joint names to be passed in construction, rather than in the part itself

**Comment by PetervDooren** [comment]
All tests are passed when all repositories are checked out to fix/side_param
see https://dev.azure.com/tue-robotics/tue_robocup/_build/results?buildId=2494&view=results

File: robot_skills/src/robot_skills/torso.py
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/torso.py:17]: Can't have a mutable default argument

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]

File: robot_skills/src/robot_skills/torso.py
Comment (PetervDooren) [reply robot_skills/src/robot_skills/torso.py]: fixed

**Review by MatthijsBurgh — APPROVED** [review]

**Pull request merged**
